packages/synapse-react-client/src/components/EntityUpload/EntityUpload.test.tsx

```diff
@@ -0,0 +1,335 @@
+    const fileInput = result.container.querySelector<HTMLInputElement>(
+      'input[type="file"][id=filesToUpload]',
+    )!
+    expect(fileInput).toBeInTheDocument()
+    expect(fileInput).not.toHaveAttribute('multiple')
+  })
+
+  it('shows uploads in progress', async () => {
+    const hookReturnValue = {
+      ...mockUseUploadFileEntitiesReturn,
+      state: 'UPLOADING',
+      activeUploadCount: 1,
+      uploadProgress: [
+        {
+          file: new File(['contents'], 'file1.txt'),
+          progress: { value: 1024 * 1024 * 50, total: 1024 * 1024 * 100 },
+          status: 'UPLOADING',
+          cancel: jest.fn(),
+          pause: jest.fn(),
+          resume: jest.fn(),
+          remove: jest.fn(),
+          failureReason: undefined,
+        },
+      ],
+    } satisfies UseUploadFileEntitiesReturn
+    mockUseUploadFileEntities.mockReturnValue(hookReturnValue)
+
+    renderComponent({
+      entityId: mockFileEntity.entity.id,
+    })
+
+    await screen.findByTestId('FileUploadProgress')
+    expect(mockFileUploadProgress).toHaveBeenLastCalledWith(
+      {
+        status: 'UPLOADING',
+        fileName: 'file1.txt',
+        totalSizeInBytes: hookReturnValue.uploadProgress[0].file.size,
+        uploadedSizeInBytes: hookReturnValue.uploadProgress[0].file.size / 2,
+        onCancel: hookReturnValue.uploadProgress[0].cancel,
+        onPause: hookReturnValue.uploadProgress[0].pause,
+        onResume: hookReturnValue.uploadProgress[0].resume,
+        onRemove: hookReturnValue.uploadProgress[0].remove,
+        errorMessage: undefined,
+      },
+      expect.anything(),
+    )
+
+    await screen.findByText('Uploading 1 Item')
+  })
+
+  it('displays a prompt to the user', async () => {
+    const hookReturnValue = {
+      ...mockUseUploadFileEntitiesReturn,
+      state: 'PROMPT_USER',
+      activePrompts: [
+        {
+          info: {
+            type: 'CONFIRM_NEW_VERSION',
+            fileName: 'file1.txt',
+            existingEntityId: 'syn123',
+          },
+          onConfirmAll: jest.fn(),
+          onConfirm: jest.fn(),
+          onSkip: jest.fn(),
+          onCancelAll: jest.fn(),
+        },
+        {
+          info: {
+            type: 'CONFIRM_NEW_VERSION',
+            fileName: 'file2.txt',
+            existingEntityId: 'syn456',
+          },
+          onConfirmAll: jest.fn(),
+          onConfirm: jest.fn(),
+          onSkip: jest.fn(),
+          onCancelAll: jest.fn(),
+        },
+      ],
+    } satisfies UseUploadFileEntitiesReturn
+    mockUseUploadFileEntities.mockReturnValue(hookReturnValue)
+
+    const { user } = renderComponent({
+      entityId: mockFileEntity.entity.id,
+    })
+
+    const dialog = await screen.findByRole('dialog')
+    within(dialog).getByText('Update existing file?')
+    within(dialog).getByText(
+      'A file named "file1.txt" (syn123) already exists in this location. Do you want to update the existing file and create a new version?',
+    )
+
```

Comment:
May want to add a quick check before click to verify baseline (for each of these checks):
 expect(hookReturnValue.activePrompts[0].onConfirm).toHaveBeenCalledTimes(0)